logging/logging.go

```diff
@@ -27,6 +28,11 @@ func Setup(logLevel string) error {
 	}
 	log.SetLevel(level)
 	log.SetFormatter(&textFormatter{})
+	hook, err := promrus.NewPrometheusHook() // Expose number of log messages as Prometheus metrics.
```

Comment:
Please add to the documentation on this function that it cannot be called more than once, or bracket this section with a test to avoid causing an `error duplicate metrics collector registration attempted` (or adding multiple instances of the same hook below)

Reply:
Another solution could be to unregister the counter before registering it, in which case initialising multiple times will not cause any issue:
```
func NewPrometheusHook() (*PrometheusHook, error) {
[...]
	// Try to unregister the counter vector, in case already registered for some reason,
	// e.g. double initialisation/configuration done by mistake by the end-user.
	prometheus.Unregister(counterVec)
	// Try to register the counter vector:
	err := prometheus.Register(counterVec)
[...]
}
```
Test:
```
func TestRegisteringHookMultipleTimesShouldBeSafe(t *testing.T) {
	log.AddHook(promrus.MustNewPrometheusHook())
	log.AddHook(promrus.MustNewPrometheusHook()) // Initialising twice in a row should NOT raise any error.

	httpServePrometheusMetrics(t)

	lines := httpGetMetrics(t)
	assert.Equal(t, 0, countFor(t, log.InfoLevel, lines))

	log.Info("this is at info level!")

	lines = httpGetMetrics(t)
	assert.Equal(t, 1, countFor(t, log.InfoLevel, lines))

	log.AddHook(promrus.MustNewPrometheusHook()) // Initialising again here should NOT raise any error, but will reset counters.

	lines = httpGetMetrics(t)
	assert.Equal(t, 0, countFor(t, log.InfoLevel, lines))
}
```
@bboreham, do you think this could be a better solution?

Reply:
Since we should prob. not discuss this here, issue opened weaveworks/promrus#5 and PR ready for review weaveworks/promrus#6.